### PR TITLE
{lib}[gcccuda/2020a] NCCL v2.7.8

### PR DESCRIPTION
**Pull request by ccoulombe**
Adding latest NCCL with Cuda 11 pulling source from the redist url.

### CODE REVIEW

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/aff3f814a41afa11530e45be7485cf6d for a full test report.

**Comment by smoors** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
@smoors Kenneth brought up the point that, maybe this can be moved up to SYSTEM level. Does this actually compile anything?

**Comment by ccoulombe** [comment]
@Micket @smoors 
It skip builds so indeed it could be at SYSTEM level.

**Comment by boegel** [comment]
Maybe @bartoldeman can clarify why we have this at the `gcccuda` toolchain in the current NCCL easyconfigs.
Does it really need `CUDA(core)` as a dep?

**Comment by Micket** [comment]
Recap of slack conversation:
Yes this NCCL is tied to a particular CUDA version (11.0).

One suggested option is to introduce a new toolchain and addition to HMNS and put it there. I'm not very keen on that, seems like a lot of work for a questionable benefit to me.

Another suggested option is to create a system level easyconfig, with a dependency on CUDAcore, with a `versionsuffix = '-CUDA-11.0.2'`. I prefer this.

**Comment by bartoldeman** [comment]
it can indeed be at the system level but then you need to add a versionsuffix like in
https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/n/NCCL/NCCL-2.2.13-CUDA-9.2.148.1.eb
(and the dep is then CUDAcore instead of CUDA now)

File: easybuild/easyconfigs/n/NCCL/NCCL-2.7.8-gcccuda-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+easyblock = "MakeCp"
+
+name = 'NCCL'
+version = '2.7.8'
```
Comment (Micket) [inline easybuild/easyconfigs/n/NCCL/NCCL-2.7.8-gcccuda-2020a.eb:4]: Please add the versionsuffix  `-CUDA-%(cudaver)s`

File: easybuild/easyconfigs/n/NCCL/NCCL-2.7.8-gcccuda-2020a.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+toolchain = {'name': 'gcccuda', 'version': '2020a'}
+toolchainopts = {'optarch': 'False'}
+
+local_cuda_version = '11.0'
+
+# Download from https://developer.nvidia.com/nccl/nccl-download (after log in) or use the redist url
+source_urls = ['https://developer.download.nvidia.com/compute/redist/nccl/v%(version_major_minor)s']
+sources = ['%%(namelower)s_%%(version)s-1+cuda%s_%%(arch)s.txz' % local_cuda_version]
+checksums = ['34000cbe6a0118bfd4ad898ebc5f59bf5d532bbf2453793891fa3f1621e25653']
+
```
Comment (Micket) [inline easybuild/easyconfigs/n/NCCL/NCCL-2.7.8-gcccuda-2020a.eb:19]: Please add a dependency to `CUDAcore` 11.0.2

File: easybuild/easyconfigs/n/NCCL/NCCL-2.7.8-gcccuda-2020a.eb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+easyblock = "MakeCp"
+
+name = 'NCCL'
+version = '2.7.8'
+
+homepage = 'https://developer.nvidia.com/nccl'
+description = """The NVIDIA Collective Communications Library (NCCL) implements multi-GPU and multi-node collective
+communication primitives that are performance optimized for NVIDIA GPUs."""
+
+toolchain = {'name': 'gcccuda', 'version': '2020a'}
```
Comment (Micket) [inline easybuild/easyconfigs/n/NCCL/NCCL-2.7.8-gcccuda-2020a.eb:10]: Please move to `toolchain = SYSTEM`

**Review by Micket — CHANGES_REQUESTED** [review]
Summarizing the changes as discussed

**Comment by verdurin** [comment]
@ccoulombe @bartoldeman any update on this one?

**Comment by migueldiascosta** [comment]
will be superseded by #11803, any objections?

**Comment by boegel** [comment]
Should this be closed now that #11803 is merged?

**Comment by bartoldeman** [comment]
yes this can be closed then

**Pull request closed without merging**